src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -434,13 +434,65 @@ private static async Task<Solution> MergeDocumentChangesAsync(Solution originalS
+
+            public Document Document => ReferenceLocation.Document;
+        }
+
+        private static async Task<ImmutableArray<LocationForAffectedSymbol>> FindReferenceLocationsForSymbol(
             Document document, ISymbol symbol, CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            var builder = ArrayBuilder<LocationForAffectedSymbol>.GetInstance();
+            try
+            {
+                var referencedSymbols = await FindReferencesAsync(symbol, document, cancellationToken).ConfigureAwait(false);
+                builder.AddRange(referencedSymbols
+                    .Where(refSymbol => refSymbol.Definition.Equals(symbol))
+                    .SelectMany(refSymbol => refSymbol.Locations)
+                    .Select(location => new LocationForAffectedSymbol(location, isReferenceToExtensionMethod: false)));
+
+                // So far we only have references to types declared in affected namespace. We also need to 
+                // handle invocation of extension methods (in reduced form) that are declared in those types. 
+                // Therefore additional calls to find references are needed for those extension methods.
+                // This will returns all the references, not just in the reduced form. But we will
+                // not further distinguish the usage. In the worst case, those references are redundant because
+                // they are already covered by the type references found above.
+                if (symbol is INamedTypeSymbol typeSymbol && typeSymbol.MightContainExtensionMethods)
+                {
+                    foreach (var methodSymbol in typeSymbol.GetMembers().OfType<IMethodSymbol>())
+                    {
+                        if (methodSymbol.IsExtensionMethod)
+                        {
+                            var referencedMethodSymbols = await FindReferencesAsync(methodSymbol, document, cancellationToken).ConfigureAwait(false);
+                            builder.AddRange(referencedMethodSymbols
+                                .SelectMany(refSymbol => refSymbol.Locations)
+                                .Select(location => new LocationForAffectedSymbol(location, isReferenceToExtensionMethod: true)));
+                        }
+                    }
+                }
 
-            var progress = new StreamingProgressCollector(StreamingFindReferencesProgress.Instance);
+                return builder.ToImmutable();
+            }
+            finally
+            {
+                builder.Free();
+            }
+        }
 
+        private static async Task<ImmutableArray<ReferencedSymbol>> FindReferencesAsync(ISymbol symbol, Document document, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            var progress = new StreamingProgressCollector(StreamingFindReferencesProgress.Instance);
```

Comment:
Why are we using the internal versions versus the public ones that just collect the progress for us?

Reply:
Isn't it the only one available in `SymbolFinder_FindReferences_Current.cs`? IIRC others are in legacy file, and I was suggested to avoid them (not supporting OOP?).

Reply:
Specifically, this comment is why I didn't use the public one

```cs
    // This file contains the legacy FindReferences APIs.  The APIs are legacy because they
    // do not contain enough information for us to effectively remote them over to the OOP
    // process to do the work.  Specifically, they lack the "current project context" necessary
    // to be able to effectively serialize symbols to/from the remote process.
```